src/apps/common/submodules/navigationWizard/navigationWizard.js

```diff
@@ -156,7 +168,29 @@ define(function(require) {
 					.on('click', function(event) {
 						event.preventDefault();
 
-						thisArg[wizardArgs.cancel](wizardArgs);
+						monster.waterfall([
+							function(waterfallCallback) {
+								if (!navigationWizardFlags.cancelConfirmation) {
+									return waterfallCallback(null, true);
+								}
+
+								monster.ui.confirm(
+									self.i18n.active().navigationWizard.cancelDialogMessage,
+									function() {
+										waterfallCallback(null, true);
+									},
+									function() {
+										waterfallCallback(null, false);
+									}
+								);
+							}
+						], function(err, result) {
```

Comment:
Please rename `result` to a more meaningful name describing the purpose of this boolean.